### PR TITLE
chore: refactor actions part 1

### PR DESCRIPTION
**Pull request by dgozman**
This separates actions from `frames.ts` and `dom.ts` and unifies selector-based and handle-based versions. The unification allows selector-based actions to ensure that target element actually matches selector when action is performed, to avoid issues with element being detached or recycled.

This change covers innerText, textContent, getAttribute, innerHTML, fill and dispatchEvent actions.

### CODE REVIEW

File: test/utils.js [hunk 1]
```diff
@@ -103,6 +103,25 @@ const utils = module.exports = {
     }
   },
 
+  registerAtomicEngine: async (name) => {
```
Comment (JoelEinbinder) [inline test/utils.js:106]: An atomic engine sounds impressive!

image

File: src/injected/injectedScript.ts [hunk 2]
```diff
@@ -157,6 +157,17 @@ export default class InjectedScript {
     });
   }
 
+  pollNode<T>(target: Node | ParsedSelector, predicate: (node: Node, progress: types.InjectedScriptProgress, continuePolling: symbol) => T | symbol): types.InjectedScriptPoll<T> {
```
Comment (JoelEinbinder) [inline src/injected/injectedScript.ts:160]: Why doesn't this return a promise? Am I missing some quirk with microtasks?

Comment (JoelEinbinder) [reply src/injected/injectedScript.ts:160]: Oh I missed `continuePolling`. Carry on.

**Review by JoelEinbinder — APPROVED** [review]
Some interesting bits in here that I hadn't read before, like checking the samespace against the w3 url. But it looks like just moves, so I'll assume it worked before and it works now.

**Comment by dgozman** [comment]
Closing for now.

**Pull request closed without merging**